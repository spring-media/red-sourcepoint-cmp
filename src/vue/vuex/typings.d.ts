import { CustomPurpose, CustomVendor } from '../../sourcepoint/typings';

export type SourcepointModuleState = {
  consentedCustomVendors: CustomVendor[];
  consentedCustomPurposes: CustomPurpose[];
  grantedVendors: string[];
  consentReady: boolean;
};
