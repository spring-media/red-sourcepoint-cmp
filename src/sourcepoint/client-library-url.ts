export const getClientLibraryUrl = (): string => `https://cdn.privacy-mgmt.com/wrapperMessagingWithoutDetection.js`;
