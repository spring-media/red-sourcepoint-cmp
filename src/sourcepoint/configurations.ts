import { ConfigurationParameters, SourcepointConfigurationObject } from './typings';

export const createConfiguration = (parameters: ConfigurationParameters): Partial<SourcepointConfigurationObject> => ({
  config: {
    accountId: parameters.accountId,
    wrapperAPIOrigin: 'https://wrapper-api.sp-prod.net/tcfv2',
    propertyId: parameters.propertyId,
    mmsDomain: 'https://message.sp-prod.net',
    events: parameters.events,
  },
});
