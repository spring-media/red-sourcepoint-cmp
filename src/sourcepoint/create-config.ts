import { Config } from './typings';

export const createConfig = (parameters: Config): Config => ({
  accountId: 0,
  wrapperAPIOrigin: 'https://wrapper-api.sp-prod.net/tcfv2',
  propertyId: 0,
  mmsDomain: 'https://message.sp-prod.net',
  ...parameters,
});
