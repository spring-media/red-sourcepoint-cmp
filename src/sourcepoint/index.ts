export * from './iab-stub-script';
export * from './client-library-url';
export * from './has-custom-consent';
export * from './privacy-manager';
export * from './create-config';
export * from './get-custom-vendor-consents';
