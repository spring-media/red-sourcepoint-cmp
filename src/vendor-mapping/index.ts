export * from './vendors';
export * from './purposes';
export * from './api';
