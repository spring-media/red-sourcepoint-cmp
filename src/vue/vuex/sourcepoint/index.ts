import { Commit, Dispatch, Module } from 'vuex';
import { getCustomVendorConsents } from '../../../sourcepoint';
import { SourcepointModuleState } from '../typings';
import { CustomPurpose, CustomVendor } from '../../../sourcepoint/typings';
import {
  configureGrants,
  configureVendorPurposeMapping,
  getGrantedVendors,
  loadVendorPurposeMapping,
} from '../../../vendor-mapping';

const moduleState: SourcepointModuleState = {
  consentedCustomVendors: [],
  consentedCustomPurposes: [],
  grantedVendors: [],
};

export const mutations = {
  setCustomVendorConsents(state: SourcepointModuleState, payload: CustomVendor[]): void {
    state.consentedCustomVendors = payload;
  },
  setCustomPurposeConsents(state: SourcepointModuleState, payload: CustomPurpose[]): void {
    state.consentedCustomPurposes = payload;
  },
  setGrantedVendors(state: SourcepointModuleState, payload: string[]): void {
    state.grantedVendors = payload;
  },
};

export const getters = {
  vendorHasConsent: (state: SourcepointModuleState) => (vendorId: string): boolean =>
    state.grantedVendors.includes(vendorId),
};

export const actions = {
  async refreshCustomVendorConsents({ commit }: { commit: Commit }): Promise<void> {
    const { consentedPurposes = [], consentedVendors = [], grants = {} } = await getCustomVendorConsents();

    configureGrants(grants);

    commit('setCustomVendorConsents', consentedVendors);
    commit('setCustomPurposeConsents', consentedPurposes);
    commit('setGrantedVendors', getGrantedVendors());
  },
  async bootstrapConsents({ dispatch }: { dispatch: Dispatch }, payload: { propertyId: number }): Promise<void> {
    await dispatch('refreshCustomVendorConsents');

    const mappings = await loadVendorPurposeMapping(payload.propertyId);

    configureVendorPurposeMapping(mappings);
  },
};

export const sourcepoint: Module<SourcepointModuleState, Record<string, unknown>> = {
  namespaced: true,
  state: moduleState,
  mutations,
  actions,
  getters,
};
