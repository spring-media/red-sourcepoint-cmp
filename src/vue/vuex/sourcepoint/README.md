# sourcepoint Vuex Module

## Effects

### reloadPageOnReject

This effect reloads the page after a vendor or purpose has been rejected.

## Example

```javascript
import Vue from 'vue';
import Vuex from 'vuex';
import { sourcepoint } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/vuex/sourcepoint';
import { reloadPageOnReject } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/vuex/sourcepoint/effects';

Vue.use(Vuex);

const store = new Vuex.Store({
  strict: true,
  state: {},
  modules: {
    sourcepoint,
  },
});

reloadPageOnReject(store);
``` 
