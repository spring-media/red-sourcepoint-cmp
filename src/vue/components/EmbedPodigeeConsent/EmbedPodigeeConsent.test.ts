import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import { EmbedPodigeeConsent } from './';
import { sourcepoint } from '../../vuex/sourcepoint';
import { VENDOR_ID_PODIGEE } from '../../../vendor-mapping';

const localVue = createLocalVue();

localVue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    sourcepoint,
  },
});

describe('EmbedPodigeeConsent component', () => {
  afterEach(() => {
    store.commit('sourcepoint/setGrantedVendors', []);
  });

  it('should render the placeholder by default', () => {
    const wrapper = mount(EmbedPodigeeConsent, {
      propsData: {
        privacyManagerId: 123,
      },
      store,
      localVue,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render the embed in case of a consented vendor', () => {
    store.commit('sourcepoint/setGrantedVendors', [VENDOR_ID_PODIGEE]);

    const wrapper = mount(EmbedPodigeeConsent, {
      propsData: {
        content: '<div>Podigee Content</div>',
        privacyManagerId: 123,
      },
      store,
      localVue,
    });

    expect(wrapper.element).toMatchInlineSnapshot(`
      <div>
        <div>
          Podigee Content
        </div>
      </div>
    `);
  });

  it('should provide an embed slot for overriding the default consent content', () => {
    store.commit('sourcepoint/setGrantedVendors', [VENDOR_ID_PODIGEE]);

    const wrapper = mount(EmbedPodigeeConsent, {
      propsData: {
        privacyManagerId: 123,
      },
      slots: {
        embed: '<div>Podigee Embed Content</div>',
      },
      store,
      localVue,
    });

    expect(wrapper.element).toMatchInlineSnapshot(`
      <div>
        Podigee Embed Content
      </div>
    `);
  });
});
