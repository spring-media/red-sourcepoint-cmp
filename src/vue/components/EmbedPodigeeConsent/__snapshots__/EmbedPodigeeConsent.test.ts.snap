// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedPodigeeConsent component should render the placeholder by default 1`] = `
<div
  class="embed-placeholder__container"
  data-vendor-id="5e717ad49a0b5040d5750815"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="80"
      viewBox="0 0 80 80"
      width="80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M30.814 16.9352C28.1511 15.2406 24.6665 17.1535 24.6665 20.3099V59.6898C24.6665 62.8462 28.1511 64.759 30.814 63.0645L61.7554 43.3745C64.2254 41.8027 64.2254 38.197 61.7554 36.6252L30.814 16.9352Z"
        stroke="#CED4DA"
        stroke-width="4"
      />
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
     An dieser Stelle findest du Inhalte aus Podigee 
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
    Um mit Inhalten aus Podigee und anderen externen Inhalten zu interagieren oder diese darzustellen, brauchen wir
    deine Zustimmung.
  
  </div>
   
  <button
    class="embed-placeholder__button"
  >
     externen Inhalt aktivieren 
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir Inhalte von Drittanbietern angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Dazu ist ggf. die Speicherung von Cookies auf
        deinem Gerät notwendig. Mehr Informationen dazu findest du
        
    <a
      class="embed-placeholder__text-link"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      hier
    </a>
    .
      
  </div>
</div>
`;
